httplib.go

```diff
@@ -65,25 +65,25 @@ func ReadError(statusCode int, re []byte) error {
 	switch statusCode {
 	case http.StatusNotFound:
 		e := NotFoundError{}
-		return Wrap(unmarshalError(&e, re))
+		return unmarshalError(&e, re)
```

Comment:
To avoid code duplication (if worth the hassle):
```go
func ReadError(statusCode int, re []byte) (err error) {
	switch statusCode {
	case http.StatusNotFound:
		err = &NotFoundError{}
	case http.StatusBadRequest:
		err = &BadParameterError{}
	case http.StatusPreconditionFailed:
		err = &CompareFailedError{}
	case http.StatusForbidden:
		err = &AccessDeniedError{}
	case http.StatusConflict:
		err = &AlreadyExistsError{}
	case statusTooManyRequests:
		err = &LimitExceededError{}
	case http.StatusGatewayTimeout:
		err = &ConnectionProblemError{}
	default:
		if statusCode < 200 || statusCode > 299 {
			return Errorf(string(re))
		}
		return nil
	}
	return unmarshalError(err, re)
}
```